_plugins/sidebar.rb

```diff
@@ -0,0 +1,195 @@
+# frozen_string_literal: false
+
+require 'jekyll'
+require 'nokogiri'
+require 'json'
+
+module Jekyll
+  class Sidebar
+    attr_accessor :hash_pre_render
+    attr_accessor :filename_with_headers
+
+    def initialize
+      @hash_pre_render = {}
+      @filename_with_headers = {}
+    end
+
+    def pre_render(page)
+      menu_order = page['menu-order'].nil? ? 0 : page['menu-order']
+      url = page['url'].gsub('index.html', '').gsub('.html', '')
+      @hash_pre_render[url] = {
+        title: page['title'],
+        url: page['url'].gsub('.html', ''),
+        name: page['name'],
+        menu_order: menu_order
+      }
+    end
+
+    def post_write(site)
+      files = []
+      Dir.glob("#{site.config['destination']}/**/*.html") do |filename|
+        doc = File.open(filename) { |f| Nokogiri::HTML(f) }
+        files.push(doc)
+        title = doc.title
```

Comment:
Codacy found an issue: [Useless assignment to variable - `title`.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-design-guide-jekyll-theme/pullRequest?prid=5659803)